### PR TITLE
[IMP] web_editor: video selector component adjustments for Knowledge

### PR DESCRIPTION
**Pull request by julien-banken**
This PR will apply the following changes:
1. Introduce a new `VideoIframe` component for the `VideoSelector` component. With that change, one can easily mock the `VideoIframe` component in a tour and replace the iframe of the video with alternative content. It will then be possible to test the whole video integration flow with a tour without relying on Third-Party services.
2. Enhance the Python video parser to include the video id and the video parameters of the url being parsed. These new information will be used for the new video behavior of Knowledge.
3. Call the callback function notifying that the video url has changed when the textarea is emptied (see: `selectMedia`).
4. Authorize non-internal users to call the `/web_editor/video_url/data` route. This will ensure that portal users will be able to integrate videos in their Knowledge articles.
5. Apply some minor UI and UX adjustments:
   - The label next to the toggle buttons will now be properly aligned.
   - The textarea to input the video url will be automatically focused.

task-3297215

---
I confirm I have signed the CLA and read the PR guidelines at www.odoo.com/submit-pr

### CODE REVIEW

**Review by abd-msyukyu-odoo — CHANGES_REQUESTED** [review]
Hello! Thanks for the pr, I mostly have some questions here :)

File: addons/web_editor/static/src/components/media_dialog/video_selector.js [hunk 1]
```diff
@@ -129,6 +137,7 @@ export class VideoSelector extends Component {
             this.state.options = [];
             this.state.platform = null;
             this.state.errorMessage = '';
+            this.props.selectMedia({});
```
[the remark this answers is not in the document]
Comment (julien-banken) [reply addons/web_editor/static/src/components/media_dialog/video_selector.js:140]: > What is the purpose of this call ?

It's to notify the other components that the provided url changes. For some reason, we don't notify the other components when the input text is emptied. Without that call, the preview is not updated when the user enters a valid video link and empty the input text.

File: addons/web_editor/static/src/components/media_dialog/video_selector.xml [hunk 2]
```diff
@@ -3,27 +3,34 @@
 
 <t t-name="web_editor.VideoOption" owl="1">
     <div class="mb-1">
-        <label class="o_switch" t-on-change="props.onChangeOption">
-            <input type="checkbox" t-att-checked="props.value"/><span/>
+        <label class="d-flex align-items-start gap-2 cursor-pointer" t-on-change="props.onChangeOption">
+            <div class="o_switch flex-shrink-0">
+                <input type="checkbox" t-att-checked="props.value"/>
+                <span/>
```
[the remark this answers is not in the document]
Comment (julien-banken) [reply addons/web_editor/static/src/components/media_dialog/video_selector.xml:9]: It's for the styling of the checkboxes. When website is not installed, we have a standard checkbox. When website is installed, we have a toggle button. That `span` ensures that the styling will be correct when website is installed. It seems to be a hack.

**Review by tde-banana-odoo — APPROVED** [review]

**Review by abd-msyukyu-odoo — APPROVED** [review]

File: addons/web_editor/controllers/main.py [hunk 3]
```diff
@@ -230,8 +230,6 @@ def update_stars(self, res_model, res_id, filename, starsId, rating):
     def video_url_data(self, video_url, autoplay=False, loop=False,
                        hide_controls=False, hide_fullscreen=False, hide_yt_logo=False,
                        hide_dm_logo=False, hide_dm_share=False):
-        if not request.env.user._is_internal():
```
Comment (tde-banana-odoo) [inline addons/web_editor/controllers/main.py:233]: Change should be motivated in commit message :)

File: addons/web_editor/static/src/components/media_dialog/video_selector.xml (hunk 2 above)
[the remark this answers is not in the document]
Comment (tde-banana-odoo) [reply addons/web_editor/static/src/components/media_dialog/video_selector.xml:9]: Had the same question, so probably worth an inline comment :)

File: addons/web_editor/static/src/components/media_dialog/video_selector.xml [hunk 4]
```diff
@@ -3,27 +3,34 @@
...
+            </div>
             <span t-esc="props.label"/>
-            <span t-if="props.description" class="small text-muted ms-auto" t-esc="props.description"/>
+            <span t-if="props.description" class="text-muted" t-esc="props.description"/>
         </label>
     </div>
 </t>
 
+<t t-name="web_editor.VideoIframe">
+    <iframe height="720" width="1280" class="o_video_dialog_iframe mw-100 mh-100 overflow-hidden shadow" allowfullscreen="allowfullscreen" frameborder="0" t-att-src="this.props.src"/>
```
Comment (tde-banana-odoo) [inline addons/web_editor/static/src/components/media_dialog/video_selector.xml:18]: :socks: could be multi lined :)

Comment (tde-banana-odoo) [reply addons/web_editor/static/src/components/media_dialog/video_selector.xml:18]: Is the fixed height / width ok even on small mobiles ?

**Review by tde-banana-odoo — COMMENTED** [review]
Zboing, few small comments while passing through :) LGTM otherwise

**Comment by tde-banana-odoo** [comment]
@robodoo delegate=@abd-msyukyu-odoo

File: addons/web_editor/static/src/components/media_dialog/video_selector.xml (hunk 4 above)
Comment (julien-banken) [reply addons/web_editor/static/src/components/media_dialog/video_selector.xml:18]: It seems like the `width` and `height` attribute are ignored. The video preview has an absolution position and fills the parent element. It's maybe an optimization: Setting those attributes will help the browser avoiding "layout shift" when loading the iframe.

Comment (abd-msyukyu-odoo) [reply addons/web_editor/static/src/components/media_dialog/video_selector.xml]: Actually the `width` and `height` attributes are overridden by the css style of the element, which specifies `100%`. Not sure about the layout shift thing, if you're convinced it is useful, we can keep the attributes, if not I'd remove them :)

**Comment by abd-msyukyu-odoo** [comment]
@robodoo r+

**Pull request closed without merging**